### PR TITLE
Prettier Config Fix

### PR DESCRIPTION
**Pull request by spider-sauce**
### Description

This PR is an effort to correct and improve the `prettier` implementation.

Changes made:
* updated `.prettierrc` to be more specific with formatting rules based on file extension.
* ignored specific sections based on relevant team requests.
### Motivation and Context

Prettier indirectly re-formatted a lot of YAML spec docs (API Fortress and saucectl), this is an effort to remedy that problem.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -10,4 +10,14 @@ build
...
+docs/web-apps/automated-testing/cucumberjs-playwright/yaml.md
+docs/web-apps/automated-testing/cypress/yaml
+docs/web-apps/automated-testing/cypress/yaml.md
+docs/web-apps/automated-testing/playwright/yaml.md
+docs/web-apps/automated-testing/puppeteer/yaml.md
+docs/web-apps/automated-testing/replay/yaml.md
+docs/web-apps/automated-testing/testcafe/yaml.md
+docs/api-testing/on-prem
+docs/api-testing/composer
+docs/api-testing/composer.md
```
Comment (lysannep) [inline .prettierignore:23]:
```suggestion
*.json
docs/web-apps/automated-testing/cucumberjs-playwright/yaml.md
docs/web-apps/automated-testing/cypress/yaml
docs/web-apps/automated-testing/playwright/yaml.md
docs/web-apps/automated-testing/puppeteer/yaml.md
docs/web-apps/automated-testing/replay/yaml.md
docs/web-apps/automated-testing/testcafe/yaml.md
docs/api-testing/on-prem
docs/api-testing/composer
docs/api-testing/composer.md
```

File: .prettierignore [hunk 2]
```diff
@@ -10,4 +10,14 @@ build
 tests
 webdriver
 *.yml
-*.json
\ No newline at end of file
+*.json
+docs/web-apps/automated-testing/cucumberjs-playwright/yaml.md
+docs/web-apps/automated-testing/cypress/yaml
+docs/web-apps/automated-testing/cypress/yaml.md
```
Comment (lysannep) [inline .prettierignore:16]:
```suggestion
```

Comment (lysannep) [reply .prettierignore:16]: @spider-sauce There are no code blocks in docs/web-apps/automated-testing/cypress/yaml.md --- only the docs/web-apps/automated-testing/cypress/yaml folder should be added to .prettierignore.

File: .prettierignore (hunk 1 above)
Comment (lysannep) [inline .prettierignore:23]: @spider-sauce I think we should also add the following folders to .prettierignore --> for the [API reference](https://docs.saucelabs.com/dev/api/) and the docs folder for the platform integrations for Backtrace (error reporting).
```suggestion
docs/api-testing/composer.md
docs/dev/api
docs/error-reporting/platform-integrations
```

File: .prettierrc [hunk 3]
```diff
@@ -23,7 +25,26 @@
...
     },
     {
-      "files": "./**/css/*.css",
+      "files": "*.{yml,yaml}",
+      "options": {
+        "parser": "yaml",
+        "tabWidth": 2,
+        "useTabs": false,
+        "semi": false,
+        "singleQuote": true
```
Comment (aliciakrebs) [inline .prettierrc:34]: should be false?

Comment (aliciakrebs) [inline .prettierrc:34]: should be false?

```suggestion
        "singleQuote": true
```
```suggestion
        "singleQuote": true
```

File: .prettierignore [hunk 4]
```diff
@@ -10,4 +10,14 @@ build
 tests
 webdriver
 *.yml
-*.json
\ No newline at end of file
+*.json
+docs/web-apps/automated-testing/cucumberjs-playwright/yaml.md
```
Comment (mhan83) [inline .prettierignore:14]: I'm a bit worried about the maintainability of this ignore list. Anyone who contributes to the docs and adds code snippets would need to know whether the file they are editing is being ignored or not. Even in this PR, I see one snippet that's inadvertently had prettifying applied.

Is there any interest in setting this prettier config: `embeddedLanguageFormatting: "off"`

They are just code snippets after all. If a reader copy-pastes the snippet, they can apply whatever prettifier they have configured locally.

**Review by LoryApiFortress — APPROVED** [review]
APIT docs: approved!

File: .prettierignore (hunk 4 above)
Comment (spider-sauce) [reply .prettierignore:14]: @mhan83 I think that's a valid option moving forward, as most of the issues are with the code blocks in general

File: .prettierignore (hunk 2 above)
Comment (spider-sauce) [reply .prettierignore]: I removed these files from the ignore list and just added the `"embeddedLanguageFormatting": "off"` to `.prettierrc`. I just tested it on one of the yaml docs and it worked.

File: .prettierrc (hunk 3 above)
Comment (spider-sauce) [reply .prettierrc:35]: for yaml files I believe single quotes are the preferred best practice. However with the change to ignoring embedded code altogether I'm not sure this matters much anymore.

File: .prettierignore (hunk 1 above)
Comment (spider-sauce) [reply .prettierignore]: I think by ignoring the embedded code languages I don't think this is necessary anymore. However we should still review/change anything in those files that needs to be updated before we merge.

File: .prettierignore (hunk 4 above)
Comment (mhan83) [reply .prettierignore]: Great. But I think its best to revert the initial prettyfier changes and re-run with the new config; there are still lots of snippets that have been improperly formatted.

Comment (lysannep) [reply .prettierignore]: ^Agreed.

**Comment by spider-sauce** [comment]
Resolved in https://github.com/saucelabs/sauce-docs/pull/1705

**Pull request closed without merging**